src/core/mac/mac_frame.hpp

```diff
@@ -1411,6 +1467,51 @@ class BeaconPayload
     ExtendedPanId mExtendedPanId;
 } OT_TOOL_PACKED_END;
 
+/**
+ * This class implements CSL IE data structure.
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+class CslIe
+{
+public:
+    /**
+     * This method returns the CSL Period.
+     *
+     * @returns the CSL Period.
+     *
+     */
+    uint16_t GetPeriod(void) const { return mPeriod; }
+
+    /**
+     * This method sets the CSL Period.
+     *
+     * @param[in]  aPeriod  The CSL Period.
+     *
+     */
+    void SetPeriod(uint16_t aPeriod) { mPeriod = aPeriod; }
```

Comment:
The byte sequence in IE header would be different depending on endian-ness of the device CPU itself

We should use `mPeriod = Encoding::LittleEndian::HostSwap16(aPeriod)` (assuming CSL IE requires little-endian). Similarly for other getters/setters than deal with `uint16_t` or `uint32_t` values.